library/spdm_requester_lib/libspdm_req_key_exchange.c

```diff
@@ -517,9 +517,12 @@ static libspdm_return_t libspdm_try_send_receive_key_exchange(
     *session_id = libspdm_generate_session_id(req_session_id, rsp_session_id);
     session_info = libspdm_assign_session_id(spdm_context, *session_id, false);
 
-    /* session_info cannot be null as the check after libspdm_allocate_req_session_id assures
-     * that there is space for a new session ID and its initialization. */
-    LIBSPDM_ASSERT(session_info != NULL);
+    if (session_info == NULL) {
```

Comment:
Only this check seems reasonable. It align with https://github.com/DMTF/libspdm/pull/1485.

Can you please slip this specific one? We can merge this one at first, then debate the rest

Reply:
> Only this check seems reasonable. It align with #1485.
> 
> Can you please slip this specific one? We can merge this one at first, then debate the rest

Done! I have updated the PR to only fix this specific one.